packages/next/next-server/server/next-server.ts

```diff
@@ -1007,6 +1014,34 @@ export default class Server {
     const isProduction = !this.renderOpts.dev
     const isDynamicPathname = isDynamicRoute(pathname)
     const didRespond = isResSent(res)
+
+    // we lazy load the staticPaths to prevent the user
+    // from waiting on them for the page to load in dev mode
+    let staticPaths = this.staticPathsCache[pathname]
+
+    if (!isProduction && hasStaticPaths) {
+      // this is the first call so we need to block since getStaticPaths
+      // has not been called yet and we don't want to inaccurately render
+      // the fallback
+      const __getStaticPaths = async () => {
+        // TODO: bubble any errors from calling this to the client
+        const paths = await this.staticPathsWorker!.loadStaticPaths(
+          this.distDir,
+          this.buildId,
+          pathname,
+          !this.renderOpts.dev && this._isLikeServerless
+        )
+        this.staticPathsCache[pathname] = paths
+        return paths
+      }
+
+      if (!staticPaths) {
+        staticPaths = await __getStaticPaths()
+      } else {
+        withCoalescedInvoke(__getStaticPaths)(`staticPaths-${pathname}`, [])
```

Comment:
Let's still always respond with this to dedupe multiple requests in a row (think: multiple tabs open in dev).